File: app/src/main/java/com/xfl/kakaotalkbot/Bridge.java
package com.xfl.kakaotalkbot;

import org.mozilla.javascript.Context;
import org.mozilla.javascript.ScriptableObject;
import org.mozilla.javascript.annotations.JSStaticFunction;

public class Bridge extends ScriptableObject {
    public Bridge() {

    }

    @JSStaticFunction
    public static ScriptableObject getScopeOf(String scriptName) {
        if (!isAllowed(scriptName)) {
            return null;
        }
        if (NotificationListener.Companion.getContainer().get(scriptName) == null) {
            //Context.reportError("java.lang.NullPointerException: The script '"+scriptName+"' is not compiled yet");

            return null;
        }
        try {
            return NotificationListener.Companion.getContainer().get(scriptName).getScope();
        } catch (Throwable e) {
            Context.reportError(e.toString());
            return null;
        }
    }

    @JSStaticFunction
    public static boolean isAllowed(String scriptName) {

        return MainApplication.getContext().getSharedPreferences("settings" + scriptName, 0).getBoolean("allowBridge", true);
    }

    public String getClassName() {
        return "Bridge";
    }


}

File: app/src/main/java/kr/oror/arabot/Bridge.kt
package kr.oror.arabot

import org.mozilla.javascript.Context
import org.mozilla.javascript.ScriptableObject
import org.mozilla.javascript.annotations.JSStaticFunction

class Bridge : ScriptableObject() {

    override fun getClassName(): String {
        return "Bridge"
    }

    companion object {
        @JvmStatic
        @JSStaticFunction
        fun getScopeOf(scriptName: String): ScriptableObject? {
            if (!isAllowed(scriptName)) {
                return null
            }
            if (ScriptsManager.container[scriptName] == null) {
                //Context.reportError("java.lang.NullPointerException: The script '"+scriptName+"' is not compiled yet");

                return null
            }
            return try {
                ScriptsManager.container[scriptName]!!.getScope()
            } catch (e: Throwable) {
                Context.reportError(e.toString())
                null
            }

        }

        @JvmStatic
        @JSStaticFunction
        fun isAllowed(scriptName: String): Boolean {

            return MainApplication.context!!.getSharedPreferences("settings$scriptName", 0).getBoolean("allowBridge", true)
        }
    }


}
